# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../htop/0001-htop-check-for-libtinfo.patch
# Copyright (C) 2013 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

check if we need to link against libtinfo to not have missing symbols

compile-time error
------------------------------------------------------------------------------
ld: htop-CRT.o: undefined reference to symbol 'nocbreak'
ld: note: 'nocbreak' is defined in DSO /lib/libtinfo.so.5 so try adding it to the linker command line
------------------------------------------------------------------------------

--- a/configure.ac	2013-12-22 02:46:17.146617569 +0100
+++ b/configure.ac	2013-12-22 02:56:00.300126717 +0100
@@ -101,6 +101,10 @@
    AC_CHECK_HEADERS([curses.h],[:],[missing_headers="$missing_headers $ac_header"])
 fi

+# ncurses can be also build as separate libraries
+# check if we need to link against libtinfo to not have missing symbols
+AC_CHECK_LIB([tinfo], [nocbreak], [], [])
+
 if test ! -z "$missing_libraries"; then
   AC_MSG_ERROR([missing libraries: $missing_libraries])
 fi
